src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/UseDefaultDllImportSearchPathsAttribute.cs

```diff
@@ -79,21 +90,25 @@ public override void Initialize(AnalysisContext context)
                     defaultValue: UnsafeBits);
                 var defaultDllImportSearchPathsAttributeOnAssembly = compilation.Assembly.GetAttributes().FirstOrDefault(o => o.AttributeClass.Equals(defaultDllImportSearchPathsAttributeTypeSymbol));
 
+                // Does not analyze local functions. To analyze local functions, we'll need to use RegisterSyntaxAction.
                 compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
                 {
-                    var symbol = symbolAnalysisContext.Symbol;
+                    var symbol = (IMethodSymbol)symbolAnalysisContext.Symbol;
 
-                    if (!symbol.IsExtern || !symbol.IsStatic)
+                    // LibraryImport methods will be partial, and the generated implementation will have a non-null PartialDefinitionPart property
+                    if (!symbol.IsStatic || symbol.PartialDefinitionPart != null)
```

Comment:
Functionally, this looks fine, but I think we should still check for the method being either `IsExtern` or `IsPartialDefinition` for performance. Otherwise, we're doing the more costly analysis on every static method that isn't a partial implementation.